### ISSUE TITLE
TypeError: BackgroundGeolocation.addWatcher is not a function

### ISSUE THREAD

**Issue by Gladiatornoshield** [capacitor-community/background-geolocation#68]
**Describe the bug**
Good Day I am having trouble implementing this solution on my Ionic 6 capacitor 3 project. After installing the plugin i follow all the steps to configure the project files. 

**To Reproduce**
I import the module inside app.component.ts
Then I paste the usage code inside initialize app. Also tried running it inside a function starttracking. 

improt used 

import {registerPlugin} from "@capacitor/core";
const {BackgroundGeolocation} = registerPlugin("BackgroundGeolocation");

**Expected behavior**
I want to capture latitude and longitude details of the device when it starts up or when they change. I am using an android simulator to test. 

**Screenshots**
![image](https://user-images.githubusercontent.com/73953775/168517531-38625004-0c8d-42f9-bbda-f2c466f4dc22.png)
![image](https://user-images.githubusercontent.com/73953775/168517638-aca89aed-19bf-4c0c-a938-d37614fd7f5a.png)

**Desktop (please complete the following information):**
 - Windows 10
 - Chrome
 - 101

**Smartphone (please complete the following information):**
 - Device: Android Simulator
 - Android Studio
 - Version 11

**Additional context**
My problem is that I am not sure if i am applying the code at the correct place. I have also tried using typescript import but give me an error saying registerPlugin cannot be found. I would really appreciate if you can just point me to the correct direction.

**Comment by Gladiatornoshield** [comment]
Good Day

I have removed them but i get error on the 

addWatcher
openSettings
removeWatcher

Property 'addWatcher' does not exist on type 'unknown'

But with the braces in place the error disappears but brings [TypeError: BackgroundGeolocation.addWatcher is not a function] when i run the app. 

![image](https://user-images.githubusercontent.com/73953775/168579685-c4409056-0ae3-48b2-8a41-d764fda17027.png)

**Comment by diachedelic** [comment]
Ah, you are using Typescript. Refer to the README for how to import the TypeScript types.

**Comment by Gladiatornoshield** [comment]
Good Day. Thank you so much for your responses. I really appreciate it. I have managed to import it as follows. 

import {registerPlugin} from "@capacitor/core";
import {BackgroundGeolocationPlugin} from "@capacitor-community/background-geolocation";
const BackgroundGeolocation = registerPlugin<BackgroundGeolocationPlugin>("BackgroundGeolocation");

Now i get required logs but I get result undefined when running on simulator. 

![image](https://user-images.githubusercontent.com/73953775/169098631-ebb8b45b-8197-4033-a7d3-802c2287427d.png)

Instead of getting the location object i see the following on the console 

![image](https://user-images.githubusercontent.com/73953775/169098905-5a55fb1a-3ead-4bf3-a0fd-b3ae66c8891d.png)

When starting up for the first time it askes for permissions. Not sure what could be causing this behaviour, But really need this to work as its the final piece of the App i have been building.

**Comment by Gladiatornoshield** [comment]
I'm still not getting that object. I have create a fresh ionic install with the following details.

![image](https://user-images.githubusercontent.com/73953775/169217029-43e318c3-7104-4b4f-a8ef-41d3b78e3524.png)

Then I configured the Android Manifest and strings XML files. Loads as expected. Asks for permissions but gives me the same results. I am attached the exact code.

import { Component } from '@angular/core';
import {registerPlugin} from "@capacitor/core";
import {BackgroundGeolocationPlugin} from "@capacitor-community/background-geolocation";
const BackgroundGeolocation = registerPlugin<BackgroundGeolocationPlugin>("BackgroundGeolocation");


@Component({
  selector: 'app-home',
  templateUrl: 'home.page.html',
  styleUrls: ['home.page.scss'],
})
export class HomePage {

  constructor() {}

  ngOnInit() {
    // To start listening for changes in the device's location, add a new watcher.
// You do this by calling 'addWatcher' with an options object and a callback. A
// Promise is returned, which resolves to the callback ID used to remove the
// watcher in the future. The callback will be called every time a new location
// is available. Watchers can not be paused, only removed. Multiple watchers may
// exist simultaneously.
BackgroundGeolocation.addWatcher(
  {
      // If the "backgroundMessage" option is defined, the watcher will
      // provide location updates whether the app is in the background or the
      // foreground. If it is not defined, location updates are only
      // guaranteed in the foreground. This is true on both platforms.

      // On Android, a notification must be shown to continue receiving
      // location updates in the background. This option specifies the text of
      // that notification.
      backgroundMessage: "Cancel to prevent battery drain.",

      // The title of the notification mentioned above. Defaults to "Using
      // your location".
      backgroundTitle: "Tracking You.",

      // Whether permissions should be requested from the user automatically,
      // if they are not already granted. Defaults to "true".
      requestPermissions: true,

      // If "true", stale locations may be delivered while the device
      // obtains a GPS fix. You are responsible for checking the "time"
      // property. If "false", locations are guaranteed to be up to date.
      // Defaults to "false".
      stale: false,

      // The minimum number of metres between subsequent locations. Defaults
      // to 0.
      distanceFilter: 50
  },
  function callback(location, error) {
      if (error) {
          if (error.code === "NOT_AUTHORIZED") {
              if (window.confirm(
                  "This app needs your location, " +
                  "but does not have permission.\n\n" +
                  "Open settings now?"
              )) {
                  // It can be useful to direct the user to their device's
                  // settings when location permissions have been denied. The
                  // plugin provides the 'openSettings' method to do exactly
                  // this.
                  BackgroundGeolocation.openSettings();
              }
          }
          return console.error(error);
      }

      return console.log(location);
  }
).then(function after_the_watcher_has_been_added(watcher_id) {
  // When a watcher is no longer needed, it should be removed by calling
  // 'removeWatcher' with an object containing its ID.
  BackgroundGeolocation.removeWatcher({
      id: watcher_id
  });
});

  }



}

**Comment by Gladiatornoshield** [comment]
![image](https://user-images.githubusercontent.com/73953775/169217833-5e906213-86f0-4350-8117-4987ff8e6e0e.png)

Thats what i see on the console. Not sure if maybe its a simulator issue or I'm just missing something. It does not seems to be running the call back as it does not give me any errors or location object

**Comment by diachedelic** [comment]
Yes, you are removing the watcher immediately after adding it. It does not get a chance to produce a location object.

**Comment by Gladiatornoshield** [comment]
Awesome. I removed the watcher. Now I get the location object. Thank you sooo much for your patience and help really appreciate it. So if the watcher is never removed would that affect the app in anyway. I want to keep it live until the app is terminated.

**Comment by diachedelic** [comment]
That should be fine, so long as your users can work out how to terminate your app.

**Comment by Gladiatornoshield** [comment]
Good Day @diachedelic Hope you doing well 

I want to save the location object to my DB but the resulting object always appears to run last on my code sequence. Where can I manipulate it so I can run updateLocation function after the resulting object has been created. I'm on ionic 6 and capacitor 3.
